play.pokemonshowdown.com/src/battle-log.ts

```diff
@@ -17,6 +17,7 @@ import type { Battle } from './battle';
 import type { BattleScene } from './battle-animations';
 import { Dex, Teams, toID, toRoomid, toUserid, type ID } from './battle-dex';
 import { BattleTextParser, type Args, type KWArgs } from './battle-text-parser';
+import type { Net } from './client-connection';
```

Comment:
Out of interest, does this work differently to jQuery in that it doesn't set the `X-Requested-With` header?

Reply:
I suppose it does. It's using raw XHR. You can just look at the source.